source/common/mongo/proxy.h

```diff
@@ -73,6 +77,31 @@ class AccessLog {
 typedef std::shared_ptr<AccessLog> AccessLogSharedPtr;
 
 /**
+ * Mongo fault configuration.
+ */
+class FaultConfig {
+public:
+  virtual ~FaultConfig() {}
+
+  virtual uint32_t delayPercent() const PURE;
+  virtual uint64_t delayDuration() const PURE;
+};
+
+class FaultConfigImpl : public FaultConfig {
+public:
+  FaultConfigImpl(const Json::Object& fault_config);
+
+  uint32_t delayPercent() const override { return delay_percent_; }
+  uint64_t delayDuration() const override { return duration_ms_; }
+
+private:
+  uint32_t delay_percent_;
```

Comment:
if I do `const` here then delay_percent_ has to be in the initializer list something like

FaultConfigImpl::FaultConfigImpl(const Json::Object& fault_config) : 
   delay_percent_(static_cast<uint32_t>(fault_config.getObject("fault")->getInteger("percent"))),
 ...

Let me know, i'm not thrilled with the construction above but could do it either way.